src/org/opendatakit/briefcase/ui/export/ExportPanel.java

```diff
@@ -16,412 +16,125 @@
-      btnChooseExportDirectory.setEnabled(false);
-      btnPemFileChooseButton.setEnabled(false);
-      pickStartDate.setEnabled(false);
-      pickEndDate.setEnabled(false);
-      btnExport.setEnabled(false);
-      // reset the termination future so we can cancel activity
-      terminationFuture.reset();
-    } else {
-      btnChooseExportDirectory.setEnabled(true);
-      btnPemFileChooseButton.setEnabled(true);
-      pickStartDate.setEnabled(true);
-      pickEndDate.setEnabled(true);
-      btnExport.setEnabled(true);
-      // touch-up with real state...
-      updateExportButton();
-      // retain progress text (to display last export outcome)
-    }
-    // remember state...
-    exportStateActive = active;
-  }
-
-  void resetExport() {
-  }
-
-  private List<String> export() {
-    return tableModel.getSelectedForms().parallelStream()
+  private List<String> export(ExportConfiguration defaultConfiguration) {
+    form.disableUI();
+    terminationFuture.reset();
+    List<String> errors = forms.getSelectedForms().parallelStream()
         .peek(FormStatus::clearStatusHistory)
         .map(formStatus -> (BriefcaseFormDefinition) formStatus.getFormDefinition())
-        .flatMap(formDefinition -> this.export(formDefinition).stream())
+        .flatMap(formDefinition -> this.export(defaultConfiguration, formDefinition).stream())
         .collect(toList());
+    form.enableUI();
+    return errors;
   }
 
-  private List<String> export(BriefcaseFormDefinition formDefinition) {
+  private List<String> export(ExportConfiguration defaultConfiguration, BriefcaseFormDefinition formDefinition) {
     List<String> errors = new ArrayList<>();
-    Optional<File> pemFile = Optional.ofNullable(pemPrivateKeyFilePath.getText()).map(File::new).filter(File::exists);
+    ExportConfiguration conf = forms.getConfiguration(formDefinition).orElse(defaultConfiguration);
+    Optional<File> pemFile = conf.mapPemFile(Path::toFile).filter(File::exists);
     if ((formDefinition.isFileEncryptedForm() || formDefinition.isFieldEncryptedForm()) && !pemFile.isPresent())
       errors.add(formDefinition.getFormName() + " form requires is encrypted and you haven't defined a valid private key file location");
     else
       try {
         ExportAction.export(
-            new File(txtExportDirectory.getText().trim()),
+            conf.mapExportDir(Path::toFile).orElseThrow(() -> new RuntimeException("Wrong export configuration")),
             ExportType.CSV,
             formDefinition,
             pemFile.orElse(null),
             terminationFuture,
-            pickStartDate.convert().getDateWithDefaultZone(),
-            pickEndDate.convert().getDateWithDefaultZone()
+            conf.mapStartDate((LocalDate ld) -> Date.from(ld.atStartOfDay(ZoneId.systemDefault()).toInstant())).orElse(null),
+            conf.mapEndDate((LocalDate ld) -> Date.from(ld.atStartOfDay(ZoneId.systemDefault()).toInstant())).orElse(null)
```

Comment:
This code is complex enough, and duplicated, so how about a method?
```
conf.mapStartDate(this::dateFromLocal).orElse(null),
conf.mapEndDate  (this::dateFromLocal).orElse(null)
```
```
private Date dateFromLocal(LocalDate ld) {
  return Date.from(ld.atStartOfDay(ZoneId.systemDefault()).toInstant());
}
```

Reply:
Now that this is on `ExportAction` class, what I'd really want to do is to migrate the uses of `Date` to `LocalDate` to avoid that ugly conversion...